bigquery/tables.js

```diff
@@ -0,0 +1,145 @@
+ * @param {string} options.bucket A Google Cloud Storage bucket to use for storage.
+ * @param {string} options.file The file to save results to within Google Cloud Storage.
+ * @param {string} options.dataset The ID of the dataset to use.
+ * @param {string} options.table The ID of the project to use.
+ * @param {string} options.format Format to export as - either 'CSV', 'JSON', or 'AVRO'.
+ * @param {boolean} [options.gzip] Optional. Whether or not data should be compressed using GZIP.
+ * @param {function} callback Callback function to receive query results.
+ */
+function exportTableToGCS (options, callback) {
+  var gcsFileObj = storage.bucket(options.bucket).file(options.file);
+
+  // Export table
+  // See https://googlecloudplatform.github.io/gcloud-node/#/docs/google-cloud/latest/bigquery/table?method=export
+  var table = bigquery.dataset(options.dataset).table(options.table);
+  table.export(
+    gcsFileObj,
+    { format: options.format, gzip: options.gzip },
+    function (err, job) {
+      if (err) {
+        return callback(err);
+      }
+
+      console.log('ExportTableToGCS: submitted job %s!', job.id);
+      return callback(null, job);
+    }
+  );
+}
+// [END export_table_to_gcs]
+
+// [START poll_export_job]
+/**
+ * Check the status of a BigQuery table export job.
+ *
+ * @param {string} jobId The ID of the export job to poll.
+ * @param {function} callback Callback function to execute when the table is exported.
+ */
+function pollExportJob (jobId, callback) {
+  var job = bigquery.job(jobId);
+  job.getMetadata(function (err, metadata) {
+    if (err) {
+      return callback(err);
+    }
+    console.log('PollExportJob: job status: %s', metadata.status.state);
+
+    // If job is done, return metadata; if not, return an error.
+    if (metadata.status.state === 'DONE') {
+      return callback(null, metadata);
+    } else {
+      return callback(new Error('Job %s is not done', jobId));
+    }
+  });
+}
+// [END poll_export_job]
+// [END complete]
+
+// The command-line program
+var cli = require('yargs');
+
+var program = module.exports = {
+  exportTableToGCS: exportTableToGCS,
+  pollExportJob: pollExportJob,
+  main: function (args) {
+    // Run the command-line program
+    cli.help().strict().parse(args).argv;
+  }
+};
+
+cli
+  .command('export <bucket> <file> <dataset> <table> [--format] [--gzip]', 'Export a table from BigQuery to Google Cloud Storage.', {}, function (options) {
+    program.exportTableToGCS(options, console.log);
+  })
+  .command('poll <jobId>', 'Check the status of a BigQuery table export job.', {}, function (options) {
+    program.pollExportJob(options.jobId, console.log);
+  })
+  .example('node $0 export sample-bigquery-export data.json github_samples natality JSON', 'Export github_samples:natality to gcs://sample-bigquery-export/data.json as JSON')
+  .example('node $0 poll job_12345ABCDE', 'Check the status of BigQuery job 12345ABCDE')
+  .options({
+    format: {
```

Comment:
These aren't global options, so you should move them into the `export` command. Here's an example of how: https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/logging/sinks.js#L168